packages/virtualized-lists/Lists/VirtualizedList.js

```diff
@@ -1542,7 +1542,10 @@ class VirtualizedList extends StateSafePureComponent<Props, State> {
       this._listMetrics.getContentLength() !== this._sentEndForContentLength
     ) {
       this._sentEndForContentLength = this._listMetrics.getContentLength();
-      onEndReached({distanceFromEnd});
+      // It will only runs when data is available and also it wont trigger onEndReached on First Mount as data will be empty
+      if (data && data.length > 0) {
```

Comment:
From a colleague of mine:

> It wont trigger onEndReached on First Mount as data will be empty

This is not true, apart from maybe in this users specific scenario.
It also is inconsistent with the logic below for `onStartReached`.
And... we can render e.g. headers

From this comment, I believe we can't really land the change (also a test is failing internally, I'm trying to figure out if that's because of this change).

In the body of the PR, you mention:

> ... will also prevent onEndReached calls on first mount thus prevent the bug for OnEndReached.

Is there an issue for the "bug for OnEndReached"? do you mind linking it here so we can match the issue with the PR?